File: frontend/src/api/getPromt.ts
import axios from "axios"

const BASE_URL = import.meta.env.VITE_BASE_URL
const PORT = import.meta.env.VITE_PORT

export const getPrompt = async (data: {
	sessionId: string
	lobbyId: string
}) => {
	try {
		const response = await axios.post(
			`${BASE_URL}:${PORT}/get-prompt`,
			data
		)
		return response.data
	} catch (error) {
		throw error
	}
}

File: api/handlers/prompt/getPromt.go
package handlers

import (
	"encoding/json"
	"net/http"

	"github.com/salvizj/Redraw/types"
	"github.com/salvizj/Redraw/utils"
)

func GetPromptHandler(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
		return
	}

	var req types.GetPromptRequest

	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		http.Error(w, "Bad request: "+err.Error(), http.StatusBadRequest)
		return
	}

	prompt, err := utils.GetPrompt(req.SessionID, req.LobbyID)
	if err != nil {
		http.Error(w, "Failed to get prompt: "+err.Error(), http.StatusInternalServerError)
		return
	}

	response := types.GetPromptResponse{
		Prompt: prompt,
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	if err := json.NewEncoder(w).Encode(response); err != nil {
		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
	}
}
